lib/screens/create_game2/create_game_map.dart

```diff
@@ -0,0 +1,75 @@
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:google_maps_flutter/google_maps_flutter.dart';
+import 'package:niira/loading.dart';
+import 'package:niira/services/location_service.dart';
+import 'package:niira/utilities/map_styles/create_game_map.dart';
+import 'package:provider/provider.dart';
+
+class CreateGameMap extends StatefulWidget {
+  @override
+  State<CreateGameMap> createState() => CreateGameMapState();
+}
+
+class CreateGameMapState extends State<CreateGameMap> {
+  final Completer<GoogleMapController> _controller = Completer();
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: FutureBuilder(
+        future: context.watch<LocationService>().getUsersCurrentLocation(),
+        builder: (context, snapshot) {
+          if (snapshot.hasData) {
+            /// Put current users location in `LatLng` object so we can use it
+            /// in flutter_google_maps objects
+            final _playerLocation = LatLng(
+              snapshot.data.latitude as double,
+              snapshot.data.longitude as double,
+            );
+
+            // On load map, show and zoom on the players location
+            final _initialCameraPosition = CameraPosition(
+              target: _playerLocation,
+              zoom: 17,
+            );
+
+            // create and set the boundary and player location icons
+            final circles = <Circle>{
```

Comment:
You could also add an extension function to create a circle from a location... I'm a broken record sorry!

Reply:
lol your allgood thanks for the tips!!

Reply:
didnt end up making the circle extension, was a bit heavy as i still needed to pass a LatLng object and create the circle with a circleId so making it looked like "Circle(circleId: CircleId('player_position_icon')).toPlayerIcon(playerLocation);"

the other extensions were super elegant and i think its alright as is. thanks again for the tip!